### PR TITLE
TKET-3469 Allow setting of RNG seed

### PR DESCRIPTION
**Pull request by doug-q**

### CODE REVIEW

**Comment by yao-cqc** [comment]
> I see that in [line 251 of seed = cast(Optional[int], kwargs.get("seed"))](https://github.com/CQCL/pytket-qiskit/blob/74208fd5e9fe2efca725a6b190b34b37dec16373/pytket/extensions/qiskit/backends/aer.py#L251) We do this differently. I would have expected each shot to use a new seed. Which is correct and why?

Do you mean each circuit should use a new seed? I guess the question is that if you process a two identical circuits in a batch should you expect the same results, and I think the answer is no.

The following code gives two different sampling outcomes, so I think it is fine. Ideally pytket-qulacs should behave the same.

```python
from pytket.extensions.qiskit.backends import AerBackend
from pytket.circuit import Circuit
b = AerBackend()
c = Circuit(1).H(0).measure_all()
d = b.get_compiled_circuit(c)
h = b.process_circuits([d,d], [10000,10000], seed=100)
res = b.get_results(h)
print(res[0].get_counts())
print(res[1].get_counts())
```

```
Counter({(1,): 5015, (0,): 4985})
Counter({(1,): 5068, (0,): 4932})
```

**Comment by doug-q** [comment]
Hmm, I don't think your example demonstrates the problem.

If I add  `print(batch_order)` after 

```
        circuit_batches, batch_order = _batch_circuits(circuits, n_shots_list)
``` 

I see 
```
[[0, 1]]
Counter({(1,): 5015, (0,): 4985})
Counter({(1,): 5068, (0,): 4932})
```

Which tells me that these circuits are batched together and there is only one call to run. How can we modify your example so that it results in multiple batches?

**Comment by yao-cqc** [comment]
You are right...
So the results produced by the following is quite questionable

```python
from pytket.extensions.qiskit.backends import AerBackend
from pytket.circuit import Circuit
b = AerBackend()
c = Circuit(1).H(0).measure_all()
d = b.get_compiled_circuit(c)
h = b.process_circuits([d,d], [10000,10001], seed=100)
res = b.get_results(h)
print(res[0].get_counts())
print(res[1].get_counts())
```

```
Counter({(1,): 5015, (0,): 4985})
Counter({(1,): 5016, (0,): 4985})
```

**Comment by cqc-alec** [comment]
> You are right... So the results produced by the following is quite questionable

Yes I would say this is a bug in pytket-qiskit.

**Comment by yao-cqc** [comment]
I would prefer seeding `process_circuits`  instead of `QulacsBackend` so it's consistent with pytket-qiskit's API. We can leave the`_sample_quantum_state` method as it is.

**Comment by doug-q** [comment]
I will update to use the parameter-to-process_circuits approach from `pytket-qisket` to inject the seed.

**Comment by doug-q** [comment]
@yao-cqc Ready for another look. 

The testing contortions are a little gross but this seems better than the alternatives I considered: code duplication or multiple inheritance.

I've gone ahead and bumped the version to 0.30.0. We could instead go to 0.29.1 but it seems we don't do that?

I will need help to do a release.

**Comment by yao-cqc** [comment]
Thanks! The code looks good to me.

@cqc-alec do we usually bump the version in a separate PR? Sorry I haven't done a release in ages so can you help Doug with the release?

**Comment by cqc-alec** [comment]
Yeah we wouldn't normally bump the version here, but do that in the release branch.

Happy to help with the release.

File: docs/changelog.rst [hunk 1]
```diff
@@ -1,6 +1,11 @@
 Changelog
 ~~~~~~~~~
 
+0.30.0 (August 2023)
```
Comment (yao-cqc) [inline docs/changelog.rst:4]:
```suggestion
Unreleased
```

Comment (yao-cqc) [reply docs/changelog.rst:4]: As suggested by Alec, we do this later in the release branch.

**Review by yao-cqc — APPROVED** [review]
Thanks!

**Pull request merged**